website/static/js/profile.js

```diff
@@ -550,13 +591,65 @@ var SocialViewModel = function(urls, modes) {
         return false;
     });
 
+    self.addWebsite = function() {
+        this.profileWebsites.push(ko.observable().extend({
+            trimmedURL: true
+            }));
+    }
+    
+    self.removeWebsite = function(profileWebsite) {
+        for (var i=0; i < self.profileWebsites().length; i++) {
+            if (profileWebsite == ko.toJS(self.profileWebsites()[i]))
+                self.profileWebsites.splice(i, 1);
+        }
+    }
+
+
+    if (self.profileWebsites().length == 0) {
+        self.addWebsite();
+    }
+
     self.fetch();
 };
 SocialViewModel.prototype = Object.create(BaseViewModel.prototype);
 $.extend(SocialViewModel.prototype, SerializeMixin.prototype, TrackedMixin.prototype);
 
-var ListViewModel = function(ContentModel, urls, modes) {
+SocialViewModel.prototype.serialize = function() {
+    var serializedData = ko.toJS(this);
+    var profileWebsites = serializedData.profileWebsites;
+    
+    function removeBlankValues(value) {
```

Comment:
Again, if we don't allow blank values to be added to the list of website, this becomes unnecessary.